### PR TITLE
Create RESOURCES.md and add resources

### PR DESCRIPTION
**Pull request by Smarticles101**
Ref. #345 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: docs/RESOURCES.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Useful Resources
+
+* [StackOverflow](http://stackoverflow.com/questions/tagged/java).
+* [The official java subreddit](https://www.reddit.com/r/java)
+* [The exercism/java Gitter room](https://gitter.im/exercism/java)
+* [Java documentation](https://docs.oracle.com/javase/8/)
+* [Java Programming Books](https://github.com/vhf/free-programming-books/blob/master/free-programming-books.md#java)
```
Comment (Smarticles101) [inline docs/RESOURCES.md]: These are resources I would use, can anyone list some more?

File: docs/RESOURCES.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Useful Resources
+
+* [StackOverflow](http://stackoverflow.com/questions/tagged/java).
+* [The official java subreddit](https://www.reddit.com/r/java)
```
Comment (stkent) [inline docs/RESOURCES.md]: Is this official?

Also, please capitalize java -> Java in the link description :)

File: docs/RESOURCES.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+* [The exercism/java Gitter room](https://gitter.im/exercism/java)
```
Comment (stkent) [inline docs/RESOURCES.md]: I'd remove this since I think these resources are mostly intended to be external?

File: docs/RESOURCES.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+* [Java documentation](https://docs.oracle.com/javase/8/)
```
Comment (stkent) [inline docs/RESOURCES.md]: This one we _can_ call the "Official Java documentation"!

**Comment by stkent** [comment]
Failure should be fixed by #674; might need to rebase once that is merged to make this pass.

**Comment by Smarticles101** [comment]
Okay. Now I need to figure out how to use git rebase. Learning time! :grin:

**Comment by Smarticles101** [comment]
Okay I think that worked

**Comment by stkent** [comment]
Sweet, thanks! LGTM.

**Pull request merged**